### PR TITLE
8280686: Remove Compile::print_method_impl

### PR DESCRIPTION
**Pull request by navyxliu**
This is a follow-up task of JDK-8280274. I think it is still reasonable to only evaluate
"CompilerPhaseTypeHelper::to_string(cpt)" when NOT_PRODUCT is defined, but we can do
that in the function body. We should avoid from messing around interfaces with macros.

Actually, it's not necessary to have print_method_impl as a public interface at all.
The only client of it is print_method().

Here is the current interface of Compile::print_method
V1: print_method(CompilerPhaseType cpt, int level)
V2: print_method(CompilerPhaseType cpt, Node* n, int level)

The only client of V2 is vector.cpp. It attempts to dump the phase name and vec_box node. It's impossible to to have n=NULL. No one will consume property <"phase :NULL">.

I think we should go in this way.
```
void Compile::print_method(CompilerPhaseType cpt, int level) {
  print_method(cpt, NULL, level);
  }
```

Testing: 
Verify that igv is still working with PrintIdealGraphLevel=3/4

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280686](https://bugs.openjdk.java.net/browse/JDK-8280686): Remove Compile::print_method_impl

### Reviewers
 * [Nils Eliasson](https://openjdk.java.net/census#neliasso) (@neliasso - **Reviewer**)
 * [Christian Hagedorn](https://openjdk.java.net/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7229/head:pull/7229` \
`$ git checkout pull/7229`

Update a local copy of the PR: \
`$ git checkout pull/7229` \
`$ git pull https://git.openjdk.java.net/jdk pull/7229/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7229`

View PR using the GUI difftool: \
`$ git pr show -t 7229`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7229.diff">https://git.openjdk.java.net/jdk/pull/7229.diff</a>

</details>

### CODE REVIEW

**Review by neliasso — APPROVED** [review]
Nice cleanup!

Looks good.

File: src/hotspot/share/opto/compile.cpp [hunk 1]
```diff
@@ -4815,29 +4815,23 @@ void Compile::sort_macro_nodes() {
 }
 
 void Compile::print_method(CompilerPhaseType cpt, int level) {
-  print_method_impl(cpt, NOT_PRODUCT(CompilerPhaseTypeHelper::to_string(cpt) COMMA) level);
+  print_method(cpt, NULL, level);
 }
 
 void Compile::print_method(CompilerPhaseType cpt, Node* n, int level) {
```
Comment (chhagedorn) [inline src/hotspot/share/opto/compile.cpp:4821]: Could we also turn `Node* n` into a parameter with default argument `NULL`? Then we can merge both `print_method` methods into one.

Comment (navyxliu) [reply src/hotspot/share/opto/compile.cpp:4821]: yes, it's possible. all we need to do is to flip the order from "Node* n, int level" to "int level, Node* n = NULL"

**Review by chhagedorn — APPROVED** [review]
That looks good to me. Thanks Xin for doing the updates!

**Review by neliasso — APPROVED** [review]
Even better! Approved.

**Review by TobiHartmann — APPROVED** [review]
Nice cleanup!

**Pull request closed without merging**